### DISCUSSION TITLE
Content allways aligning to top left in button on pointer over

### DISCUSSION THREAD

**Discussion by YWilix** [AvaloniaUI/Avalonia#11786] [Q&A] [answered]
Hello , I'am trying to align a button content to the center using a style selector (with :pointerover) this is the code : 

`       <Style Selector="Button:pointerover /template/ ContentPresenter#PART_ContentPresenter">
      <Setter Property="Background" Value="{StaticResource MainGradient2}"/>
      <Setter Property="TextBlock.Foreground" Value="White"/>
      <Setter Property="TextBlock.FontWeight" Value="Bold"/>
      <Setter Property="Width" Value="120"/>
      <Setter Property="Height" Value="40"/>
    </Style>   `

for some reason the code doesn't work even if I change it to this : 

`    <Style Selector="Button:pointerover">
      <Setter Property="Background" Value="{StaticResource MainGradient2}"/>
      <Setter Property="TextBlock.Foreground" Value="White"/>
      <Setter Property="TextBlock.FontWeight" Value="Bold"/>
      <Setter Property="Width" Value="120"/>
      <Setter Property="Height" Value="40"/>
    </Style>   `

also changing it to anything other than Center doesn't change anything
The content stays aligned to the top left of the button 
do you have any idea on how to fix this ?

**Comment by YWilix** [comment, marked as answer]
I don't know if this is a bug or not but while I am changing the code I removed the Height and Width setters and the problem was solved 
I had another style also with a Selector for all buttons . it looks like this :
`    <Style Selector="Button">
      <Setter Property="Background" Value="{StaticResource MainGradient2}"/>
      <Setter Property="Foreground" Value="White"/>
      <Setter Property="FontWeight" Value="Bold"/>
      <Setter Property="HorizontalContentAlignment" Value="Center"/>
      <Setter Property="VerticalContentAlignment" Value="Center"/>
      <Setter Property="Width" Value="120"/>
      <Setter Property="Height" Value="40"/>
    </Style>
`

**Reply by timunie** [reply]
For reference, read the styling Troubleshoot docs. For example, a locally set value has higher priority than a style defines one: https://docs.avaloniaui.net/docs/styling/troubleshooting
